### ISSUE TITLE
get number of channels in mc connections

### ISSUE THREAD

**Issue by porres** [pure-data/pure-data#1989]
This is something definitely not in the "that-would-be-nice" category as I see it. Here's a proof of concept, I am designing an output~ like abstraction for multichannel signals. I need to use [clone] just to include a simple [hip~ 3] object for all channels and I need to resize [clone] according to the number of channels and I need to get that information somehow.

I have a simple external for that right now clled [nchs~] (see picture below), it could be part of Vanilla, maybe as new [snake~] function, like [snake~ nchs] or built into [inlet~]. I like the [inlet~] idea cause it could also give us '0' for no signal connected, which also helps a lot (and it's not something I can do as an object, built in or not I guess). 

![Screen Shot 2023-06-14 at 12 54 06](https://github.com/pure-data/pure-data/assets/5360142/6beb179f-1373-4728-baa3-4a408d0930ae)

ping @millerpuckette

**Comment by Spacechild1** [comment]
I recently thought about that feature too.

>  like the [inlet~] idea cause it could also give us '0' for no signal connected, which also helps a lot (and it's not something I can do as an object, built in or not I guess).

Definitely not. Not every patch has an inlet~ in the first place. The signal might come from a receive~ or from an abstraction.

I think a snake~ method would be the obvious choice. (In fact, there are several outstanding snake~ methods that need discussion. I might open a dedicated ticket for that as a discussion place.)

**Comment by porres** [comment]
> Definitely not. Not every patch has an inlet~ in the first place. The signal might come from a receive~ or from an abstraction.

Fair point, another unrelated inlet functionality for stating if there's a connection or not would be then welcome as well... I got a very dirty and not elegant hack at all for doing this (and I'm using it in the new [output~] abstraction in vanilla) though, so I could potentially live with it for now (and miller might consider it a "that-would-be-nice" thing).

[snake~ nchs] it is then... I now see my [nchs~] external is blowing up Pd so I'm afraid I'm doing things wrong :( 

> (In fact, there are several outstanding snake~ methods that need discussion. I might open a dedicated ticket for that as a discussion place.)

We can move this there then. Anyway, I hope there's time to include this on in 0.54!

cheers

**Comment by ben-wes** [comment]
this is closed with 017bb82cd74265b6a83c900025b2874eb041655f, i assume.
